intl_translation_arb/test/extraction_test.dart

```diff
@@ -0,0 +1,487 @@
+      expect(extraction.warnings.first, '''
+Skipping invalid Intl.message invocation
+    <Intl.message(someString, name: "noVariables", desc: "Invalid. Not a literal")>
+    reason: Intl.message messages must be string literals
+    from example.dart    line: 2, column: 29''');
+    });
+
+    test('Message with escapable charactesr', () {
+      // This is unremarkable in English, but the translated versions will contain
+      // characters that ought to be escaped during code generation.
+
+      final extraction = MessageExtraction();
+      final messages = extraction.parseContent('''
+    String escapable() => Intl.message(
+      "Escapable characters here: ",
+      name: "escapable",
+      desc: "Escapable characters",
+    );
+    ''', 'example.dart');
+
+      final message = messages['escapable'];
+      expect(message.name, 'escapable');
+      expect(icuMessageToString(message), 'Escapable characters here: ');
+    });
+
+    test('Message with outer plural', () {
+      final extraction = MessageExtraction();
+      final messages = extraction.parseContent('''
+    String outerPlural(n) => Intl.plural(n,
+        zero: 'none',
+        one: 'one',
+        other: 'some',
+        name: 'outerPlural',
+        desc: 'A plural with no enclosing message',
+        args: [n]);
+    ''', 'example.dart');
+
+      final message = messages['outerPlural'];
+      expect(message.name, 'outerPlural');
+      expect(message.arguments, ['n']);
+
+      expect(icuMessageToString(message),
+          '{n,plural, =0{none}=1{one}other{some}}');
+    });
+
+    test('Message with outer gender', () {
+      final extraction = MessageExtraction();
+      final messages = extraction.parseContent('''
+    String outerGender(g) => Intl.gender(g,
+      male: 'm',
+      female: 'f',
+      other: 'o',
+      name: 'outerGender',
+      desc: 'A gender with no enclosing message',
+      args: [g]);
+    ''', 'example.dart');
+
+      final message = messages['outerGender'];
+      expect(message.name, 'outerGender');
+      expect(message.arguments, ['g']);
+
+      expect(
+          icuMessageToString(message), '{g,select, female{f}male{m}other{o}}');
+    });
+
+    test('Message with plural that fails parsing', () {
+      final extraction = MessageExtraction();
+      final messages = extraction.parseContent('''
+    String pluralThatFailsParsing(noOfThings) => Intl.plural(
+      noOfThings,
+      one: "1 thing:",
+      other: "\$noOfThings things:",
+      name: "pluralThatFailsParsing",
+      args: [noOfThings],
+      desc: "How many things are there?",
+    );
+    ''', 'example.dart');
+
+      final message = messages['pluralThatFailsParsing'];
+      expect(message.name, 'pluralThatFailsParsing');
+      expect(message.arguments, ['noOfThings']);
+      expect(icuMessageToString(message),
+          '{noOfThings,plural, =1{1 thing:}other{{noOfThings} things:}}');
+      throw 'Shouldnt this fail?';
+    });
```

Comment:
I am testing the extraction of the messages from intl_translation.
The name is pluralThatFailsParsing,  when should this fail? Extraction or generation?

Reply:
Parsing would mean extraction. From looking, it's not immediately obvious why it should fail, so it should have a comment, but presumably it fails parsing in the original.

Reply:
Looking at this in the original, this looks bad there. The message doesn't fail parsing, and is actually used in the tests.